config: remove `block_sync` key (#415)
cometbft/cometbft: docs/core/block-sync.md

## Using Block Sync

> Nodes start in Block Sync mode by default. In this mode, the CometBFT daemon

Reviewer: ```suggestion
When starting from scratch, nodes will use the Block Sync mode.
In this mode, the CometBFT daemon
```
